[CORE-945] Add content to RefArch usage section (#815)
gruntwork-io/docs: _docs-sources/refarch/usage/maintain-your-refarch/undeploying.md

```

## Manually undeploying multiple modules or an entire environment

_If you are absolutely sure you want to run destroy on multiple modules or an entire environment_, you can use the `destroy-all` command. For example, to undeploy the entire staging environment, you'd run:

:::danger

This operation cannot be undone!

:::

```bash
cd stage
terragrunt destroy-all
```

Terragrunt will then run `terragrunt destroy` in each subfolder of the current working directory, processing them in
reverse order based on the dependencies you define in the `terragrunt.hcl` files.

To avoid interactive prompts from Terragrunt (use very carefully!!), add the `--terragrunt-non-interactive` flag:

```bash
cd stage
terragrunt destroy-all --terragrunt-non-interactive
```

To undeploy everything except a couple specific sub-folders, add the `--terragrunt-exclude-dir` flag. For example, to
run `destroy` in each subfolder of the `stage` environment except MySQL and Redis, you'd run:

```
cd stage
terragrunt destroy-all \
    --terragrunt-exclude-dir stage/us-east-1/stage/data-stores/mysql \
    --terragrunt-exclude-dir stage/us-east-1/stage/data-stores/redis
```

## Removing the Terraform state

> **NOTE: Deleting state means that you lose the ability to manage your current Terraform resources! Be sure to only
> delete once you have confirmed all resources are destroyed.**

Reviewer: ```suggestion
:::danger

Deleting state means that you lose the ability to manage your current Terraform resources! Be sure to only delete once you have confirmed all resources are destroyed.

:::
```

Once all the resources for an environment have been destroyed, you can remove the state objects managed by `terragrunt`.

> The reference architecture manages state for each environment in an S3 bucket in each environment's AWS account.

Reviewer: ```suggestion
The Reference Architecture manages state for each environment in an S3 bucket in each environment's AWS account.
```

    prevent you from cleanly calling destroy twice.
  - To address these instabilities, it is recommended to destroy the resources in groups. For example, you can start
    by destroying all the services first (e.g., `stage/REGION/stage/services`), then the data stores (e.g.,
    `stage/REGION/stage/data-stores`), and finally the networking resources (e.g., `stage/REGION/stage/networking`).
  - When identifying groups to destroy, use [terragrunt
    graph-dependencies](https://terragrunt.gruntwork.io/docs/reference/cli-options/#graph-dependencies) to view the
    dependency graph so that you destroy the modules in the right order.

- **Empty + Delete S3 buckets using the web console (when destroying whole environments).**
  - As mentioned in [Pre-requisite: force_destroy on S3 buckets](#pre-requisite-force_destroy-on-s3-buckets), it is
    recommended to set `force_destroy = true` prior to running destroy so that Terraform can destroy the S3 buckets.
    However, this can be cumbersome if you are destroying whole environments, as it can be difficult to flip the bit in
    every single module.

>   - Instead, oftentimes it is faster and more convenient to first empty and then delete the buckets using the AWS web console prior to
>     invoking `destroy` with `terragrunt`.

Reviewer: ```suggestion
  - Alternatively, it is often faster and more convenient to empty and delete the buckets using the AWS web console before executing the `destroy` command with `terragrunt`.
```
